src/nn/Modules/Dropout.cpp

```diff
@@ -0,0 +1,35 @@
+/*******************************************************
+ * Copyright (c) 2017, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+#include <af/autograd/Functions.hpp>
+
+#include <af/nn/Init.hpp>
+#include <af/nn/Modules/Dropout.hpp>
+
+namespace af
+{
+    namespace nn
+    {
+        using namespace autograd;
+
+        Dropout::Dropout(dim4 shape, double drop_ratio)
+        {
+            mask = nn::lecunUniform(shape, f32, false) > drop_ratio;
+        }
+
+        Dropout::Dropout(const Variable &m) :
+            mask(m)
+        {
+        }
+
+        Variable Dropout::forward(const Variable &input)
+        {
+            return mask * input;
```

Comment:
I think drop out uses a new mask for every forward pass. Use `nn::uniform()` for the mask. Then call `mask.setCalcGrad(false)` because you don't need to calculate the gradient of the dropout.

And this is only performed during training. Can you modify Module.hpp to contain a `bool m_train` ? Set it to false by default in `Module::Module`, but set it to true in `Module::train`, and false in `Module::eval`